modin/experimental/batch/pipeline.py

```diff
@@ -0,0 +1,396 @@
+                    ]
+        return partitions
+
+    def compute_batch(
+        self,
+        postprocessor: Optional[Callable] = None,
+        pass_partition_id: Optional[bool] = False,
+        pass_output_id: Optional[bool] = False,
+    ):
+        """
+        Run the completed pipeline + any postprocessing steps end to end.
+
+        Parameters
+        ----------
+        postprocessor : Callable, default: None
+            A postprocessing function to be applied to each output partition.
+            The order of arguments passed is `df` (the partition), `output_id`
+            (if `pass_output_id=True`), and `partition_id` (if `pass_partition_id=True`).
+        pass_partition_id : bool, default: False
+            Whether or not to pass the numerical partition id to the postprocessing function.
+        pass_output_id : bool, default: False
+            Whether or not to pass the output ID associated with output queries to the
+            postprocessing function.
+
+        Returns
+        -------
+        list or dict or DataFrame
+            If output ids are specified, a dictionary mapping output id to the resulting dataframe
+            is returned, otherwise, a list of the resulting dataframes is returned.
+        """
+        if len(self.outputs) == 0:
+            ErrorMessage.single_warning(
+                "No outputs to compute. Returning an empty list. Please specify outputs by calling `add_query` with `is_output=True`."
+            )
+            return []
+        if not self.is_output_id_specified and pass_output_id:
+            raise ValueError(
+                "`pass_output_id` is set to True, but output ids have not been specified. "
+                + "To pass output ids, please specify them using the `output_id` kwarg with pipeline.add_query"
+            )
+        if self.is_output_id_specified:
+            outs = {}
+        else:
+            outs = []
+        modin_frame = self.df._query_compiler._modin_frame
+        partitions = modin_frame._partition_mgr_cls.row_partitions(
+            modin_frame._partitions
+        )
+        for i, node in enumerate(self.outputs):
+            partitions = self._complete_nodes(node.operators + [node], partitions)
+            for part in partitions:
+                part.drain_call_queue(num_splits=1)
+            output_partitions = partitions
+            if postprocessor:
+                output_partitions = []
+                for partition_id, partition in enumerate(partitions):
+                    args = []
+                    if pass_output_id:
+                        args.append(node.output_id)
+                    if pass_partition_id:
+                        args.append(partition_id)
+                    output_partitions.append(
+                        partition.add_to_apply_calls(postprocessor, *args)
+                    )
+            else:
+                output_partitions = [
```

Comment:
Quick note - here we add a identity function if there's no post processing, and then block partition the output Dataframes, since its not likely they'll be re-block partitioned otherwise. Is this ok to keep?